src/index.ts

```diff
@@ -81,22 +86,139 @@ export class Client {
       headers
     };
 
-    return fetchImpl(url, request).then(response => {
-      return response.json().then(
-        (body: SubmissionBody): SubmissionResponse => {
-          return { body, response };
+    // first check if we need to add the stripe paymentMethod
+    if (this.stripePromise && opts.createPaymentMethod) {
+      // Get Stripe payload
+      const payload = await opts.createPaymentMethod();
+
+      if (payload.error) {
+        // Return the error in case Stripe failed to create a payment method
+        return {
+          response: null,
+          body: {
+            errors: [
```

Comment:
does the server not provide this payload itself? If so, can we just return the response from the server which should be correctly formatted like so

Reply:
This is not a server side response, this is from Stripe itself.